src/ray/util/event.cc

```diff
@@ -148,26 +148,51 @@ void EventManager::ClearReporters() { reporter_map_.clear(); }
 ///
 thread_local std::unique_ptr<RayEventContext> RayEventContext::context_ = nullptr;
 
+std::unique_ptr<RayEventContext> RayEventContext::first_instance_ = nullptr;
+
+std::atomic<int> RayEventContext::first_instance_started_setting_(0);
+
+std::atomic<bool> RayEventContext::first_instance_finished_setting_(false);
+
 RayEventContext &RayEventContext::Instance() {
   if (context_ == nullptr) {
     context_ = std::unique_ptr<RayEventContext>(new RayEventContext());
   }
   return *context_;
 }
 
+RayEventContext &RayEventContext::OriginContext() {
```

Comment:
could this be private? we should hide all these logic (along with IsInitialized()) behind `RayEventContext::Instance()`

Reply:
Rename to a private method `GlobalInstance` and make `RayEvent` to be friend class.